runtime/ts/entity.ts

```diff
@@ -51,11 +59,14 @@ export class Entity {
     }
     this[Symbols.identifier] = id;
   }
+
   toLiteral() {
     return this.rawData;
   }
 
-  static get type() {
+  abstract get key(): {tag: string, schema: Schema};
```

Comment:
This is a static getter in schema generated subclasses. You won't be able to maintain this as an abstract method once you return the static signifier, so there's no point this class being converted to abstract either.